python/cudf/cudf/core/column/column.py

```diff
@@ -1360,6 +1339,31 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             elif np.issubdtype(data.dtype, np.datetime64):
                 data = data.fillna(np.datetime64("NaT"))
 
+    elif hasattr(arbitrary, "__array_interface__"):
+        # CUDF assumes values are always contiguous
+        desc = arbitrary.__array_interface__
+        shape = desc["shape"]
+        arb_dtype = np.dtype(desc["typestr"])
+        # CUDF assumes values are always contiguous
+        if len(shape) > 1:
+            raise ValueError("Data must be 1-dimensional")
+
+        if desc["strides"] is not None:
+            arbitrary = np.ascontiguousarray(arbitrary)
+
+        if dtype is not None:
+            arbitrary = arbitrary.astype(dtype)
+
+        if arb_dtype.kind == "M":
+            data = datetime.DatetimeColumn.from_numpy(arbitrary)
```

Comment:
Is this guaranteed to work against `non-numpy` inputs? Should probably just remove this function and make this work with the general `as_column` factory.

Reply:
Yes, instead we can do this: 
```
data = as_column(pa.array(arbitrary), nan_as_null=nan_as_null)
```